rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -548,18 +555,33 @@ public String joinLRA(URL lraId, Class<?> resourceClass, URI baseUri,
         return null;
     }
 
-    @Override
-    public URL updateCompensator(URL recoveryUrl, URL compensateUrl, URL completeUrl,
-                                 URL forgetUrl, URL statusUrl,
-                                 String compensatorData) throws GenericLRAException {
-        return null;// TODO
+    public URL updateCompensator(URL url, Class<?> aClass, URI uri, String s) throws GenericLRAException {
+        return null; // TODO
     }
 
-    @Override
-    public void leaveLRA(URL lraId, String compensatorUrl) throws GenericLRAException {
+    public void leaveLRA(URL lraId, String body) throws GenericLRAException {
         Response response = null;
 
-        lraTracef(lraId, "leaving LRA, compensator url: %s", compensatorUrl);
+        try {
+            aquireConnection();
+
+            response = getTarget().path(String.format(leaveFormat, getLRAId(lraId.toString())))
+                    .request()
+                    .header(LRA_HTTP_HEADER, lraId)
+                    .put(Entity.entity(body, MediaType.TEXT_PLAIN));
+
+            if (Response.Status.OK.getStatusCode() != response.getStatus()) {
+                LRALogger.i18NLogger.error_lraLeaveUnexpectedStatus(response.getStatus(), response);
+                throw new GenericLRAException(null, response.getStatus(), "", null);
```

Comment:
I think if possible the exception should contain some text for easier tracking the error.

btw. maybe a doubt for me is, if it is necessary to log error when exception is thrown just at the place

Reply:
Same comment as before, so I will just raise a separate JIRA.

Reply:
please, would be possible to fix the error messages just as part of this fix? it's better to discuss about the error message meaning now when context is more obvious around than some time after when the jira like JBTM-3089 will be handled as one bundle of changes.

btw. I think hte JBTM-3089 is really reasonable what it states, that's good to have it

Reply:
Same comment as above

Reply:
I think we can get the message from the log statement? This is new code so it would seem reasonable to take the same log message: LRALogger.i18NLogger.error_lraLeaveUnexpectedStatus(response.getStatus(), response) - maybe I misunderstood?

Reply:
@tomjenkinson Other than a change of the name of the first parameter to the exception constructor the code is unchanged:

-                throw new GenericLRAException(lraId, response.getStatus(), "", null);
+               throw new GenericLRAException(compensatorUrl, response.getStatus(), "", null);

ie this is not new code. It is copied directly from the other leaveLRA on line 596 in the next change set below. I agree that a text message is needed, just not in the PR (I raised JBTM-3089 to add the extra text).

Reply:
Ah - got you!